src/Tribe/Views/V2/View.php

```diff
@@ -2145,4 +2154,92 @@ protected function get_container_data() {
+
+		/**
+		 * Filters Whether the Recent Past Events Should Show for all Views.
+		 *
+		 * @since TBD
+		 *
+		 * @param aboolean $show      If we should display Recent Past Events.
+		 * @param string   $view_slug The current view slug.
+		 * @param View     $instance  The current View object.
+		 */
+		$show = apply_filters( 'tribe_events_views_v2_show_recent_past_events_view', $show, $this->get_slug(), $this );
+
+		/**
+		 *  Filters Whether the Recent Past Events Should Show for a specific View.
+		 *
+		 * @since TBD
+		 *
+		 * @param boolean $show     If we should display Recent Past Events.
+		 * @param View    $instance The current View object.
+		 */
+		$show = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_show_recent_past_events_view", $show, $this );
+
+		return $show;
+	}
+
+
+	/**
+	 * @param array $events
+	 * @param array $template_vars
+	 */
+	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
+		// todo: flatten the events if required (month, week)
+		// todo add threshold and filter
+		if ( 0 === count( $events ) && ! empty( $template_vars['show_recent_past'] ) ) {
+			$template_vars['show_recent_past'] = true;
+			$recent_past_view                  = new Recent_Past_View();
+			$recent_past_view->set_context( $this->context );
+			$instance = $recent_past_view->make( Recent_Past_View::Class );
+			$recent_past_view->template = new Template( $instance );
+			$template_vars['recent_past_view_data'] = $recent_past_view->get_template_vars();
+
+			$whitelist                              = [
+				'components/loader',
+				'components/json-ld-data',
+				'components/data',
+				'components/before',
+				'components/messages',
+				'components/breadcrumbs',
+				'components/events-bar',
+				'components/breadcrumbs',
+				'components/top-bar/today',
+				'components/top-bar/actions',
+				'components/events-bar/views/list',
+				'components/events-bar/search/keyword',
+				'components/events-bar/search/submit',
+				'components/events-bar/search-button',
+				'components/events-bar/tabs',
+				'components/events-bar/search',
+				'components/events-bar/filters',
+				'components/events-bar/views',
+				'components/events-bar/views/list/item',
+				'list/top-bar',
+				'list/top-bar/nav',
+				'list/top-bar/datepicker',
+				'list',
+				// todo: add here the templates we need and use in the recent past events view.
+			];
+			add_filter( 'tribe_template_done', static function ( $done = null, $name ) use ( $whitelist ) {
+				if ( in_array( $name, $whitelist, true ) ) {
+					return $done;
+				}
+
+				return '';
+			}, 1, 2 );
+
+			//todo hook into "tribe_template_html:components/messages" start filtering templates
+			//apply_filters( "tribe_template_html:$hook_name", $html, $file, $name, $this )
+
+			//todo hook into "tribe_template_html:month" - render
+			add_filter( 'tribe_template_html:events/v2/list', static function ( $html ) use ( $recent_past_view ) {
```

Comment:
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 76dd987a0f707a72e7a5cf0c80c21ecb</sub>